AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/InitialSync/InitialSyncOperationEvent.swift

```diff
@@ -0,0 +1,25 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+import AWSPluginsCore
+
+enum InitialSyncOperationEvent {
+    /// This event is used to notify `ModelSyncedEventEmitter` the `SyncType` of `ModelType`
+    case started(modelType: Model.Type, syncType: SyncType)
```

Comment:
Appreciate the efforts for adding comments. Feel free to take/revise/ignore my suggestions. The attempt is to remove some of the implementation details of ModelSyncedEventEmitter like "recordReceived"

```suggestion
    /// Published at the start of sync query (full or delta) for a particular Model, used by `SyncedEventEmitter` and `ModelSyncedEmitted`
    case started(modelType: Model.Type, syncType: SyncType)
```